### ISSUE TITLE
Commander Core "Pump Failure" at 0 RPM

### ISSUE THREAD

**Issue by Asylian1** [EvanMulawski/FanControl.CorsairLink#122]
Hi - been using FanControl for a while now, but just noticed that for whatever reason with this fresh install of Windows 11 and an update to the pump firmware, it seems that whenever I run FanControl I'm getting a pump failure message (which is untrue the pump seems to be just fine) with red LEDs flashing etc.. 

Is there a way around this? Do I need to factory reset the Commander Core attached?

**Comment by EvanMulawski** [comment]
Hi @Asylian1, where do you see the pump failure message? Can you post a screenshot/picture and the CorsairLink.log file (located in the Fan Control directory)?

**Comment by Asylian1** [comment]
Sure, here's the log and a picture of the pump failure icon showing on the pump:
[CorsairLink.log](https://github.com/EvanMulawski/FanControl.CorsairLink/files/13326906/CorsairLink.log)
![Screenshot_20231111_191809_Photos](https://github.com/EvanMulawski/FanControl.CorsairLink/assets/77643816/bbbaf91f-925f-46a1-bfe8-99c90a1a68a1)

**Comment by Asylian1** [comment]
So update - I did reconstruct the profile I had set from the ground up and reproduced the issue. It seems when I employ my fan curve that involves putting the fans connected to one of the cores to 0%, that this then causes the AIO to flip out and cause this issue. As how to fix not sure yet. @EvanMulawski

**Comment by Asylian1** [comment]
Another update - so this seems to only happen when the pump control specifically is set to 0%. Before update the metric was fine so not sure why it causes the AIO to sneeze an error like it's doing. 

I just manually set the pump to 0% to confirm this theory and it did indeed induce the pump failure message. Is this just normal expected behavior for the pump itself or something in FanControl you think?

Here's logs: 
[CorsairLink.log](https://github.com/EvanMulawski/FanControl.CorsairLink/files/13326963/CorsairLink.log)

**Comment by EvanMulawski** [comment]
Ah, that explains it. The device must think it's pump failure if the pump RPM is zero - since you can't "officially" set the pump to 0 RPM in iCUE.

Does it occur if you set the pump to 1%? And does it exit from pump failure status if you increase it after setting it to 0%?

**Comment by Asylian1** [comment]
Yea it still occurs at 1%, although that still registers at 0 rpm. After the error is triggered seems I can no longer control the fans and in order to clear said error I have to do a full shut down.

**Comment by Asylian1** [comment]
Actually it is fine at 1%, this time it's at around 600 rpm. Think the ui bugged a bit when I did it before. No errors yet.

**Comment by Asylian1** [comment]
Ohh and another thing - it's the INSTANT I select 0% on manual control even if there are still rpms left going.

So it doesn't seem that the pump needs to go to 0 rpm in order to trigger the error, it automatically throws one the instant it gets a 0% command regardless of current rpm.

**Comment by EvanMulawski** [comment]
When you are able, can you determine the lowest percentage that can be set before the pump turns off? Please follow these steps:

1. Set the `FANCONTROL_CORSAIRLINK_DEBUG_LOGGING_ENABLED` [environment variable](https://github.com/EvanMulawski/FanControl.CorsairLink/issues/78) to `1`.
2. Restart Fan Control.
3. Set the pump control to manual.
4. Set the pump speed to 100%.
5. Wait 10 seconds. 
6. Set the pump speed to 50%.
7. Wait 10 seconds.
8. Set the pump speed to 35%.
9. Wait 10 seconds.
10. Begin decreasing the pump speed by 1%, waiting a few seconds between each change, until you encounter the pump failure status.
11. Set the pump control/speed back to whatever you want/use.
12. Post the `CorsairLink.log` file.
13. Restart your PC to regain pump control.

With this information, I will be able to set a minimum value for the pump to prevent this from occurring. Thanks!

**Comment by UDPSendToFailed** [comment]
Just making a comment as I have also experimented with lowering the pump speed in idle to make it quieter a while ago. 

In my case with a H170i Elite LCD (non-XT), it's around 44% (~1540 RPM) where the pump starts making a grinding noise probably due to the lack of waterflow, just a guess as I'm not an expert on the mechanics of Corsair AIOs. It gets louder with lowering the speed below 40% (~1410 RPM), reducing it even further makes the pump slow down with horrible noises to end up at ~600 RPM at 20%. 

Going any lower to the 1-19% range seems to have no effect, the pump keeps running at ~600 RPM, and setting it to 0% stops it completely. Also I have noticed that if I don't decrease the speed gradually but set it to between 1-19% with one click, then it ends up in a lowered speed of the initial RPM, for example it goes from ~2750 RPM to ~1700 RPM if the starting point was 100%.
